File: Android/_Viper View_/Code/user_interface/VIPERPresenter.java
//
//  __MODULENAME__Presenter.java
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface;

import androidx.appcompat.app.AppCompatActivity;

import NAMESPACE.modules._VIPER_.contracts.VIPERContracts;

public class VIPERPresenter implements VIPERContracts.Presenter, VIPERContracts.InteractorOutput {

    public VIPERContracts.View view;
    public VIPERContracts.Interactor interactor = new VIPERInteractor(this);
    public VIPERContracts.Router router;

    public VIPERPresenter(VIPERContracts.View view) {
        this.view = view;
		this.router = new VIPERRouter((AppCompatActivity) this.view);
    }

    @Override
    public void unload() {
        this.interactor.unload();
        this.router.unload();
        this.interactor = null;
        this.router = null;
        this.view = null;
    }
}

File: Kotlin/_VIPER View_/Code/user_interface/VIPERPresenter.kt
//
//  __MODULENAME__Presenter.kt
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface

import NAMESPACE.modules._VIPER_.contracts.ConfiguracaoContracts
import androidx.appcompat.app.AppCompatActivity

class VIPERPresenter (view: VIPERContracts.View?): VIPERContracts.Presenter,
    VIPERContracts.InteractorOutput {
    var view: VIPERContracts.View? = null
    var interactor: VIPERContracts.Interactor? = VIPERInteractor(this)
    var router: VIPERContracts.Router? = null

    init {
        this.view = view
        router = VIPERRouter(view as AppCompatActivity?)
    }

    override fun unload() {
        interactor?.unload()
        router?.unload()
        interactor = null
        router = null
        view = null
    }
}
